compiler/src/dotty/tools/dotc/parsing/JavaParsers.scala

```diff
@@ -812,6 +824,17 @@ object JavaParsers {
       else
         List()
 
+
+    def permittedSubclassesOpt(isSealed: Boolean) : List[Tree] =
+      if in.token == PERMITS && !isSealed then
+        syntaxError(em"A type declaration that has a permits clause should have a sealed modifier")
+      if in.token == PERMITS then
+        in.nextToken()
+        repsep(() => typ(), COMMA)
+      else
+        // JEP-409: Class/Interface may omit the permits clause
```

Comment:
I think this is incorrect, when I run Javac on a sealed class it required the permits clause